src/test/java/de/friday/test/support/rules/dsl/specification/RuleSpecification.java

```diff
@@ -64,19 +64,51 @@ public interface RuleSpecification extends RuleRunner {
      * </pre>
      * This runs the TODOsRule against the ok.gs source code file and return the list of issues found.
      * </p>
+     *
      * @param ruleClass Source code file to run the rule against
      * @return The Rule specification
      */
     RuleSpecification whenCheckedAgainst(Class<? extends BaseGosuRule> ruleClass);
 
+    /**
+     * Specify the parameter that will be used.
+     * <p>
+     * E.g.
+     * <pre>
+     * given("TODOsRule/ok.gs").withParameter("Max","10").whenCheckedAgainst(TODOsRule.class).then().issuesFound().areEmpty();
+     * </pre>
+     * This runs the TODOsRule with parameter against the ok.gs source code file and return the list of issues found.
+     * </p>
+     *
+     * @param key   param Key
+     * @param value param Vaule
+     * @return The Rule specification
+     */
+    RuleSpecification withParameter(String key, String value);
```

Comment:
How about `withRuleProperty`? 
This way it will follow the same nomenclature of the Sonar Plugin API `@RuleProperty`.

Reply:
reimplemented